components/create-collective/sections/ConnectGithub.js

```diff
@@ -0,0 +1,406 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Flex, Box } from '@rebass/grid';
+import { URLSearchParams } from 'universal-url';
+import { FormattedMessage, defineMessages, injectIntl } from 'react-intl';
+
+import StyledButton from '../../StyledButton';
+import StyledCheckbox from '../../StyledCheckbox';
+import { P, H1 } from '../../Text';
+import GithubRepositories from './GithubRepositories';
+import StyledInputField from '../../StyledInputField';
+import Loading from '../../Loading';
+import GithubRepositoriesFAQ from '../../faqs/GithubRepositoriesFAQ';
+import { withUser } from '../../UserProvider';
+import MessageBox from '../../MessageBox';
+
+import { Router } from '../../../server/pages';
+import { getGithubRepos } from '../../../lib/api';
+import { getWebsiteUrl, getErrorFromGraphqlException } from '../../../lib/utils';
+import { LOCAL_STORAGE_KEYS, getFromLocalStorage } from '../../../lib/local-storage';
+
+class ConnectGithub extends React.Component {
+  static propTypes = {
+    query: PropTypes.object,
+    token: PropTypes.string,
+    loadingLoggedInUser: PropTypes.bool,
+    LoggedInUser: PropTypes.object,
+    refetchLoggedInUser: PropTypes.func,
+    createCollectiveFromGithub: PropTypes.func,
+    intl: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.messages = defineMessages({
+      guidelines: { id: 'openSourceApply.guidelines', defaultMessage: 'guidelines' },
+      'tos.label': {
+        id: 'createcollective.tosos.label',
+        defaultMessage: 'I agree with the {tososlink}',
+        values: {
+          tososlink: (
+            <a href="#" target="_blank" rel="noopener noreferrer">
+              terms of fiscal sponsorship
+            </a>
+          ),
+        },
+      },
+      header: { id: 'createCollective.header.create', defaultMessage: 'Create a Collective' },
+      openSourceSubtitle: {
+        id: 'collective.subtitle.opensource',
+        defaultMessage: 'Open source projects are invited to join the Open Source Collective fiscal host.',
+      },
+      repoHeader: {
+        id: 'collective.header.pickarepo',
+        defaultMessage: 'Pick a repository',
+      },
+      back: {
+        id: 'createCollective.link.back',
+        defaultMessage: 'Back',
+      },
+    });
+    this.state = {
+      result: {},
+      loadingRepos: false,
+      repositories: [],
+      checked: false,
+      error: null,
+    };
+  }
+
+  async componentDidMount() {
+    const { token } = this.props;
+    if (!token) {
+      return;
+    }
+    this.setState({ loadingRepos: true });
+
+    try {
+      const repositories = await getGithubRepos(token);
+      if (repositories.length !== 0) {
+        const filteredRepos = repositories.filter(repo => repo.stargazers_count >= 100);
```

Comment:
Do we need to do this filtering? There's already an API filter for that in https://github.com/opencollective/opencollective-api/blob/c8beb9551f03cfef27d3479feb9b6ad8a341e207/server/controllers/connectedAccounts.js#L212.

Reply:
Interesting - on localhost it was loading every single repo of mine, not just ones with over 100 stars